packages/api-request-builder/src/create-service.js

```diff
@@ -116,13 +118,22 @@ export default function createService (
       const uri =
         (withProjectKey ? `/${options}` : '') +
         endpoint +
+        // TODO this can lead to invalid URIs as getIdOrKey can return
+        //   "/?customerId", so there can be multiple questionmarks,
+        // same for when `queryParams` and `version` are present
         getIdOrKey(this.params) +
```

Comment:
does `getIdOrKey` throw an error at least if both `id` and `key` are undefined or null?

Reply:
No it does not.

```js
function getIdOrKey (params: Object): string {
  if (params.id)
    return `/${params.id}`
  else if (params.key)
    return `/key=${params.key}`
  else if (params.customerId)
    return `/?customerId=${params.customerId}`
  return ''
}
```

I opened https://github.com/commercetools/nodejs/pull/338 to keep track of this.

Reply:
It should not throw, those are optional params but the function is always called.